zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -248,6 +249,138 @@ def _get_services(cls):
+                                current_release >= bionic_train)
+
+    def tearDown(self):
+        """Cleanup loadbalancers if there are any left over."""
+        super(NeutronGatewayStatusActionsTest, self).tearDown()
+        if not self.SKIP_LBAAS_TESTS:
+            load_balancers = self.neutron_client.list_loadbalancers().get(
+                'loadbalancers', [])
+            for lbaas in load_balancers:
+                self.neutron_client.delete_loadbalancer(lbaas['id'])
+
+    def _assert_result_match(self, action_result, resource_list,
+                             resource_name):
+        """Assert that action_result contains same data as resource_list."""
+        # make sure that action completed successfully
+        self.assertEqual(action_result.status, 'completed')
+
+        # extract data from juju action
+        action_data = action_result.data.get('results', {}).get(resource_name)
+        resources_from_action = json.loads(action_data)
+
+        # pull resource IDs from expected resource list and juju action data
+        expected_resource_ids = {resource['id'] for resource in resource_list}
+        result_resource_ids = {resource['id'] for resource in
+                               resources_from_action}
+
+        # assert that juju action returned expected resources
+        self.assertEqual(result_resource_ids, expected_resource_ids)
+
+    def test_get_status_routers(self):
+        """Test that get-status-routers reports correct neutron routers."""
+        # fetch neutron routers using neutron client
+        ngw_unit = zaza.model.get_units(self.application_name,
+                                        model_name=self.model_name)[0]
+        routers_from_client = self.neutron_client.list_routers().get(
+            'routers', [])
+
+        if not routers_from_client:
+            self.fail('At least one router must be configured for this test '
+                      'to pass.')
+
+        # fetch neutron routers using juju-action
+        result = zaza.model.run_action(ngw_unit.entity_id,
+                                       'get-status-routers',
+                                       model_name=self.model_name,
+                                       action_params={"format": "json"})
+
+        # assert that data from neutron client match data from juju action
+        self._assert_result_match(result, routers_from_client, 'router-list')
+
+    def test_get_status_dhcp(self):
+        """Test that get-status-dhcp reports correct DHCP networks."""
+        # fetch DHCP networks using neutron client
+        ngw_unit = zaza.model.get_units(self.application_name,
+                                        model_name=self.model_name)[0]
+        networks_from_client = self.neutron_client.list_networks().get(
+            'networks', [])
+
+        if not networks_from_client:
+            self.fail('At least one network must be configured for this test '
+                      'to pass.')
```

Comment:
As above; should this test be skipped if there is no resource to test against rather than failing?